### PR TITLE
[FLINK-28456] Document streaming query page

### PR DESCRIPTION
**Pull request by JingsongLi**

### CODE REVIEW

File: docs/content/docs/development/query-table.md [hunk 1]
```diff
@@ -26,49 +26,67 @@ under the License.
...
 - `=`
-- `<>`
 - `<`
 - `<=`
 - `>`
 - `>=`
-- `in`
-- starts with `like`
+- `IN (...)`
+- `LIKE '%abc'`
```
Comment (LadyForest) [inline docs/content/docs/development/query-table.md:54]: IIUC, we only support start pattern?

File: docs/content/docs/development/query-table.md [hunk 2]
```diff
@@ -26,49 +26,67 @@ under the License.
...
+- `IS NULL`
+
+Table Store will sort the data by primary key, which is good for point queries
```
Comment (LadyForest) [inline docs/content/docs/development/query-table.md:57]:
```suggestion
Table Store will sort the data by primary key, which speeds up the point
and range queries. When using a composite primary key, the query filters should form a [leftmost prefix](https://dev.mysql.com/doc/refman/5.7/en/multiple-column-indexes.html)
of the primary key for good acceleration.
```

File: docs/content/docs/development/query-table.md [hunk 3]
```diff
@@ -26,49 +26,67 @@ under the License.
...
+```
+
+The primary key is catalog_id + order_id. Good acceleration is obtained
+by specifying a range filter for the leftmost prefix of the primary key.
+
+```sql
+SELECT * FROM orders WHERE catalog_id=1025;
 
-## Real-time Streaming Consumption
+SELECT * FROM orders WHERE catalog_id=1025 AND order_id=29495;
```
Comment (LadyForest) [inline docs/content/docs/development/query-table.md:79]: format the sql
```suggestion
SELECT * FROM orders WHERE catalog_id = 1025 AND order_id = 29495;
```

File: docs/content/docs/development/query-table.md [hunk 4]
```diff
@@ -26,49 +26,67 @@ under the License.
...
 
-By default, data is only visible after the checkpoint, which means
-that the streaming reading has transactional consistency.
+SELECT * FROM orders
+  WHERE catalog_id=1025
+  AND order_id>2035 AND order_id<6000;
```
Comment (LadyForest) [inline docs/content/docs/development/query-table.md:83]:
```suggestion
SELECT * FROM orders
  WHERE catalog_id = 1025
  AND order_id > 2035 AND order_id < 6000;
```

File: docs/content/docs/development/query-table.md [hunk 5]
```diff
@@ -26,49 +26,67 @@ under the License.
...
+```
+
+However, there will not be a good acceleration in the following query.
 
-Immediate data visibility is configured via
-`log.consistency` = `eventual`.
+```sql
+SELECT * FROM orders WHERE order_id=29495;
```
Comment (LadyForest) [inline docs/content/docs/development/query-table.md:89]:
```suggestion
SELECT * FROM orders WHERE order_id = 29495;
```

File: docs/content/docs/development/query-table.md [hunk 6]
```diff
@@ -26,49 +26,67 @@ under the License.
...
+```sql
+CREATE TABLE orders (
+    catalog_id BIGINT,
+    order_id BIGINT,
+    .....,
+    PRIMARY KEY (catalog_id, order_id) NOT ENFORCED
+)
+```
+
+The primary key is catalog_id + order_id. Good acceleration is obtained
```
Comment (LadyForest) [inline docs/content/docs/development/query-table.md:73]:
```suggestion
The query obtains a good acceleration by specifying a range filter for the leftmost prefix of the primary key.
```

File: docs/content/docs/development/query-table.md [hunk 7]
```diff
@@ -26,49 +26,67 @@ under the License.
...
+of the primary key for good acceleration.
+
+Suppose that a table has the following specification:
+
+```sql
+CREATE TABLE orders (
+    catalog_id BIGINT,
+    order_id BIGINT,
+    .....,
+    PRIMARY KEY (catalog_id, order_id) NOT ENFORCED
```
Comment (LadyForest) [inline docs/content/docs/development/query-table.md:69]:
```suggestion
    PRIMARY KEY (catalog_id, order_id) NOT ENFORCED -- composite primary key
```

File: docs/content/docs/development/streaming-query.md [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+# Streaming Query
+
+Currently, only Flink supports streaming query.
```
Comment (LadyForest) [inline docs/content/docs/development/streaming-query.md:29]:
```suggestion
Currently, Table Store only supports Flink streaming queries.
```

File: docs/content/docs/development/streaming-query.md [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+
+The Table Store is streaming batch unified, you can read full
+and incremental data depending on the runtime execution mode:
+
+```sql
+-- Batch mode, read latest snapshot
```
Comment (LadyForest) [inline docs/content/docs/development/streaming-query.md:35]: default batch read is latest?

File: docs/content/docs/development/streaming-query.md [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+
+## Streaming Query on Files
+
+You can incrementally consume tables directly on the lake store files. This mode has
+a lower cost compared to Kafka, but the latency will be bigger, depending on the
+checkpoint interval of the writing stream job.
+
+By default, the downstream streaming consumption is a disordered (ordered within the key)
+stream of upsert data. If you expect an ordered CDC data stream, you can configure it
+as follows (recommended):
```
Comment (LadyForest) [inline docs/content/docs/development/streaming-query.md:79]:
```suggestion
You can choose to comsume incremental changes directly from the lake store files under streaming mode. This mode has
a lower cost compared to Kafka but has a higher latency, depending on the
checkpoint interval of the writing job.

By default, the downstream streaming consumption is disordered (ordered within the key)
stream of upsert data. If you expect an ordered CDC data stream, you can configure it
as follows (recommended):
```

File: docs/content/docs/development/streaming-query.md [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+CREATE TABLE T (...)
+WITH (
+    'changelog-file' = 'true',
+    'log.changelog-mode' = 'all'
+)
+```
+
+## Streaming Query on Kafka
+
+You can configure the Kafka topic for the table, and the data written will be
```
Comment (LadyForest) [inline docs/content/docs/development/streaming-query.md:91]: For a table configuring a log system like Kafka, data will be double written to the file storage and the topic under streaming mode. For queries, there will be hybrid reads will from incremental snapshots.

**Review by LadyForest — CHANGES_REQUESTED** [review]
Thanks for your contribution! The PR is almost good to merge except for some minor comments.

File: docs/content/docs/development/query-table.md (hunk 1 above)
Comment (JingsongLi) [reply docs/content/docs/development/query-table.md:54]: Yes, this is the start pattern

File: docs/content/docs/development/query-table.md (hunk 3 above)
Comment (JingsongLi) [reply docs/content/docs/development/query-table.md:79]: When I take a look to https://dev.mysql.com/doc/refman/5.7/en/multiple-column-indexes.html
We don't have to add ` ` in here.

File: docs/content/docs/development/query-table.md (hunk 2 above)
Comment (JingsongLi) [reply docs/content/docs/development/query-table.md:57]: `speeds up the` looks good to me.
But for `the query filters should form`, I think it is not a `should`, it is just an optimization.

File: docs/content/docs/development/streaming-query.md (hunk 9 above)
Comment (JingsongLi) [reply docs/content/docs/development/streaming-query.md:35]: Yes, the comparison with streaming is highlighted here

**Review by LadyForest — APPROVED** [review]
thanks for the update，LGTM～

**Pull request merged**